vcpkg/src/lib.rs

```diff
@@ -398,19 +398,196 @@ fn find_vcpkg_target(cfg: &Config, target_triplet: &TargetTriplet) -> Result<Vcp
+                "pkg-config file {} has bogus name",
+                path.to_string_lossy()
+            ))))
+        .to_string_lossy();
+        // Read through the file and gather what we want.
+        let mut file = try!(File::open(path)
+            .map_err(|_| Error::VcpkgInstallation(format!("Couldn't open {}", path.display()))));
+        let mut pc_file_contents = String::new();
+
+        try!(file
+            .read_to_string(&mut pc_file_contents)
+            .map_err(|_| Error::VcpkgInstallation(format!("Couldn't read {}", path.display()))));
+        PcFile::from_str(&id, &pc_file_contents, &vcpkg_target.target_triplet)
+    }
+    fn from_str(id: &str, s: &str, target_triplet: &TargetTriplet) -> Result<Self, Error> {
+        let mut libs = Vec::new();
+        let mut deps = Vec::new();
+
+        for line in s.lines() {
+            // We could collect alot of stuff here, but we only care about Requires and Libs for the moment.
+            if line.starts_with("Requires:") {
+                let requires_args = line.split(":").skip(1).next().unwrap_or("").split(",");
+                for mut dep in requires_args {
+                    // Drop any versioning requirements, we only care about library order and rely upon
+                    // port dependencies to resolve versioning.
+                    if let Some(inequality) = dep.find(|c| c == '=' || c == '<' || c == '>') {
+                        dep = dep.split_at(inequality).0;
+                    }
+                    dep = dep.trim();
+                    // Sometimes Requires: lines don't have commas, so split again.
+                    deps.extend(dep.split_whitespace().map(|s| s.to_owned()));
+                }
+            } else if line.starts_with("Libs:") {
+                let lib_flags = line
+                    .split(":")
+                    .skip(1)
+                    .next()
+                    .unwrap_or("")
+                    .split_whitespace();
+                for lib_flag in lib_flags {
+                    if lib_flag.starts_with("-l") {
+                        // reconstruct the library name.
+                        let lib = format!(
+                            "{}{}.{}",
+                            if target_triplet.strip_lib_prefix {
+                                "lib"
+                            } else {
+                                ""
+                            },
+                            lib_flag.trim_left_matches("-l"),
+                            target_triplet.lib_suffix
+                        );
+                        libs.push(lib);
+                    }
+                }
+            }
+        }
+
+        Ok(PcFile {
+            id: id.to_string(),
+            libs: libs,
+            deps: deps,
+        })
+    }
+}
+
+/// Collection of PcFile.  Can be built and queried as a set of .pc files.
+#[derive(Debug)]
+struct PcFiles {
+    files: BTreeMap<String, PcFile>,
```

Comment:
Does this need to be a `BTreeMap` or can it just be a `HashMap`? I don't think the ordering of the keys ought to matter for the use case.

Reply:
Changed to HashMap